File: ui/task-tracker/src/app/shared/model/userWspStatusChangeModel.ts

import { UserStatusChangeType } from '../enums/user-status-change-type'

export class UserWspStatusChangeModel {
    public id!: number;

    /** Юзер, которого приглашают или удаляют из рабочего пространства*/
    public userId!: number;

    /** Юзер, который приглашает или удаляет из рабочего пространства */
    public requestCreatorId!: number;

    /** Дата, когда был создан запрос в UTC */
    public date!: string;

    /** Подтвердил юзер приглашение или нет */
    public requestStatus!: UserStatusChangeType;

    /** Скрыл ли юзер с фронта этот запрос(пока не используется, функционал на будущее) */
    public isHidden!: boolean;

    /** Имя рабочего пространства, куда приглашают юзера */
    public workSpaceName!: string;

    /** Фамилия, имя либо никнейм того, кто приглашает в воркспейс */
    public inviterName!: string;

    /** Эмейл того, кто приглашает в воркспейс */
    public inviterEmail!: string;

    /** Фамилия, имя либо никнейм директора воркспейса */
    public directorWspName!: string;

    /** Фамилия, имя либо никнейм юзера, которого приглашают */
    public userName!: string;

    /** Эмейл юзера, которого приглашают */
    public userEmail!: string;
}

File: TaskTracker.Core/src/Models/ResponseModels/UserWspStatusChangeModel.cs
using TaskTracker.Core.src.Enums;

namespace TaskTracker.Core.src.Models.ResponseModels
{
    public class UserWspStatusChangeModel
    {
        /// <summary>
        /// Юзер, которого приглашают или удаляют из рабочего пространства
        /// </summary>
        public int UserId { get; set; }

        /// <summary>
        /// Юзер, который приглашает или удаляет из рабочего пространства
        /// </summary>
        public int RequestCreatorId { get; set; }

        /// <summary>
        /// Дата, когда был создан запрос в UTC
        /// </summary>
        public string Date { get; set; } = string.Empty;

        /// <summary>
        /// Подтвердил юзер приглашение или нет
        /// </summary>
        public UserStatusChangeType RequestStatus { get; set; }

        /// <summary>
        /// Скрыл ли юзер с фронта этот запрос(пока не используется, функционал на будущее)
        /// </summary>
        public bool IsHidden { get; set; }

        /// <summary>
        /// Имя рабочего пространства, куда приглашают юзера
        /// </summary>
        public string WorkSpaceName { get; set; } = null!;

        /// <summary>
        /// Фамилия, имя либо никнейм того, кто приглашает в воркспейс
        /// </summary>
        public string InviterName { get; set; } = null!;

        /// <summary>
        /// Эмейл того, кто приглашает в воркспейс
        /// </summary>
        public string InviterEmail { get; set; } = null!;

        /// <summary>
        /// Фамилия, имя либо никнейм директора воркспейса
        /// </summary>
        public string DirectorWspName { get; set; } = null!;

        /// <summary>
        /// Фамилия, имя либо никнейм юзера, которого приглашают
        /// </summary>
        public string UserName { get; set; } = null!;

        /// <summary>
        /// Эмейл юзера, которого приглашают
        /// </summary>
        public string UserEmail { get; set; } = null!;
    }
}
